pkg/visitors/assemble.go

```diff
@@ -133,6 +134,15 @@ func (v *Assemble) Visit(f uefi.Firmware) error {
 		// TODO: handle the whole header instead of doing this
 		binary.LittleEndian.PutUint64(fBuf[32:], f.Length)
 
+		// Write the correct GUID to the correct spot
+		if v.UseFFS3 && f.FileSystemGUID == *uefi.FFS2 {
```

Comment:
Is the opposite also true? `if !v.UseFFS3 && r.FileSystemGUID == *uefi.FFS3 { f.FilesystemGUID = *uefi.FFS2; ... }`

Reply:
I think it can be fine even if we don't downgrade. I would leave this until we see a case where we actually need to downgrade.